pkg/daemon/ceph/client/filesystem.go

```diff
@@ -179,69 +179,22 @@ func IsMultiFSEnabled() bool {
-	//               [2] https://tracker.ceph.com/issues/23172
-
-	// * Noted section 1 - See note at top of function
-	fsAtStart, errAtStart := GetFilesystem(context, clusterName, fsName)
-	// collect information now, but don't check error yet
-	// * End of Noted section 1
 
 	// Always tell Ceph to set the new max_mds value
 	args := []string{"fs", "set", fsName, "max_mds", strconv.Itoa(int(activeMDSCount))}
 	if _, err := NewCephCommand(context, clusterName, args).Run(); err != nil {
 		return errors.Wrapf(err, "failed to set filesystem %s num mds ranks (max_mds) to %d", fsName, activeMDSCount)
 	}
-
-	// ** Noted section 2 - See note at top of function
-	// Now check the error to see if we can even determine whether we should reduce or not
-	if errAtStart != nil {
-		return errors.Wrapf(errAtStart, `failed to get filesystem %s info needed to ensure mds rank can be changed correctly,
-if num active mdses (max_mds) was lowered, USER should deactivate extra active mdses manually`,
-			fsName)
-	}
-	if int(activeMDSCount) > fsAtStart.MDSMap.MaxMDS {
-		return nil // No need to deactivate mdses if we are raising max_mds
-	}
-	logger.Debugf("deactivating some running mdses for filesystem %s", fsName)
-	// Deactivate all mdses except desired number (N); arbitrarily choose first N to live
-	fs, err := GetFilesystem(context, clusterName, fsName)
-	if err != nil {
-		logger.Warningf(
-			fmt.Sprintf("Failed to get filesystem %s info needed to deactivate running mdses. ", fsName) +
-				"using slightly stale info, this could (rarely) result in momentary loss of filesystem availability" +
-				fmt.Sprintf(": %v", err),
-		)
-		fs = fsAtStart // <-- Do the best we can to disable mdses that were active when we started
-		// Effects of stale info should be non-destructive, & unlikely the info is actually bad
-	}
-	// Deactivate any mdses with a higher rank than the desired max rank
-	// Ceph only allows mdses to be deactivated in reverse order starting with the highest rank
-	for gid := int(len(fs.MDSMap.In)) - 1; gid >= int(activeMDSCount); gid-- {
-		if err := deactivateMdsWithRetry(context, gid, clusterName, fsName); err != nil {
-			logger.Warningf("in luminous this is non-ideal but not necessarily critical: %v", err)
-		}
-	}
-	// ** End of noted section 2
-
 	return nil
 }
 
-func deactivateMdsWithRetry(context *clusterd.Context, mdsGid int, namespace, fsName string) error {
-	retries := 10
-	retrySleep := 5 * time.Second
-	var err error
-	for i := 1; i <= retries; i++ {
-		args := []string{"mds", "deactivate", fmt.Sprintf("%s:%d", fsName, mdsGid)}
-		if _, err = NewCephCommand(context, namespace, args).Run(); err == nil {
-			logger.Infof("successfully disabled mds with rank %d on attempt %d", mdsGid, i)
-			return nil
-		}
-		time.Sleep(retrySleep)
+//ReduceMDS sets the max_mads to 1 just before upgrading the MDS
+func ReduceMDS(context *clusterd.Context, clusterName, fsName string, activeMDSCount int32) error {
```

Comment:
This seems to be doing the exact same thing as the preexisting `SetNumMDSRanks`. Am I missing something? If possible, use the preexisting function.

Reply:
It's doing the same thing, as the `SetNumMDSRanks` but the reason I chose to create it as a new function is that its sole purpose is to just reduce the MDS value to 1 and nothing else.

Also, It would be a good practice to have a separate function just for that?.
If you check [this ](https://docs.ceph.com/docs/nautilus/cephfs/upgrading/#upgrading-the-mds-cluster) it reduces the MDS to 1.

WDYT?.